### ISSUE TITLE
Overflow error on distributed choice 

### ISSUE THREAD

**Issue by joey-coleman** [symphonytool/symphony#169]
> The following bug was [originally reported](https://sourceforge.net/p/compassresearch/bugs/169/) on Sourceforge by [*anonymous](https://sourceforge.net/u/*anonymous/profile/), 2013-11-26 18:15:53.700000:

channels
 on : nat
The interpreter can't manage distributed choice.  The process below causes a StackOverflowError.

process Spec = 
 begin
 state  

 all_nodes_nat : set of nat := {0,1,2}

 actions

```
NEWAlive = alive : set of nat @ 
  ([] i in set (alive) @ on!i -> NEWAlive(alive))

NEWAliveTwo = alive : set of nat @ 
  (on!1 -> NEWAliveTwo(alive union {1}))
   []
   (on!2 -> NEWAliveTwo(alive))
  [] 
  (on!0 -> NEWAliveTwo(alive))
```

  @ 
   NEWAlive(all_nodes_nat)  -- begin with no nodes on and no time passed  
 end

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/169/#9313) by [lausdahl](https://sourceforge.net/u/lausdahl/profile/), 2013-11-29 23:10:52.798000:
- **Module**: unknown --> interpreter

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/169/#949e) by [lausdahl](https://sourceforge.net/u/lausdahl/profile/), 2013-12-02 11:42:33.600000:
- **status**: open --> closed
- **assigned_to**: Anders Kaels Malmos

**Comment by joey-coleman** [comment]
> [Comment](https://sourceforge.net/p/compassresearch/bugs/169/#a04a) by [lausdahl](https://sourceforge.net/u/lausdahl/profile/), 2013-12-02 11:42:34.168000:

fixed in [85c04a23187aaf2b439aee8bbbec2c16021ca572]
